### PR TITLE
Feature/basic chat bot

### PR DESCRIPTION
**Pull request by edwinreuleneto**
Neste pull request,  estou implementando o comportamento básico para o BOT RUBENS. 

O que deve ser analisado: 
- Pergunta para o bot
- Resposta do bot
- Performance do código 

**Obs.: Este é meu primeiro código baseado em React. Se fiz algo de muito errado, me perdoe e me de um feedback :)**

### CODE REVIEW

File: src/components/chat/eventUser/index.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import React, {Component} from 'react'
+import Message from '../message/'
+
+export default class EventUserChat extends Component{
+  constructor(props) {
```
Comment (horaciosystem) [inline src/components/chat/eventUser/index.js]: Using `babel-plugin-transform-class-properties` you don't need to create the `constructor` neither do the bind for every function.
```
state = {
  userMessage: '',
  chat: []
}
 handleChangeUserMessage = (e) => {
   //...
 }
```
IMO, much more clean.

File: src/components/chat/eventUser/index.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  handleChangeUserMessage(e){
+    this.setState( {userMessage: e.target.value});
+  }
+  sendMessage(){ 
+    const message = { text : this.state.userMessage, type: 'user' };
+    console.log(message);
+    console.log(JSON.stringify(message))
+
+    // Adicionando mensagem do usuário no array de chaat atual
+    this.setState((prevState, props) => ({
```
Comment (horaciosystem) [inline src/components/chat/eventUser/index.js]: You're not using the `props` parameter here, then you can remove that and keep only the `prevState`

File: src/components/chat/eventUser/index.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    };
+
+    this.handleChangeUserMessage = this.handleChangeUserMessage.bind(this);
+    this.sendMessage = this.sendMessage.bind(this);
+  }
+  handleChangeUserMessage(e){
+    this.setState( {userMessage: e.target.value});
+  }
+  sendMessage(){ 
+    const message = { text : this.state.userMessage, type: 'user' };
```
Comment (horaciosystem) [inline src/components/chat/eventUser/index.js]: You can use a constant for type:  'user'

File: src/components/chat/eventUser/index.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    }).then(result => {
+      // Adicionando mensagem do Rubens no array de chaat atual
+      console.log('result' , result);
+      return this.setState((prevState, props) => ({
+        chat: prevState.chat.concat(result.message)
+      }));
+    })
+    .catch(err => {
+      // trata se alguma das promises falhar
+      console.error('Failed retrieving information', err);
```
Comment (horaciosystem) [inline src/components/chat/eventUser/index.js]: You can create a state property called `error` and set that when the fetch fails.
On render, you can make a partial render of the error message using this error property.

File: src/components/chat/eventUser/index.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    });
+  }
+  render(){
+    return (
+      <div className="userField">
+        {
+          this.state.chat.map(message =>
```
Comment (horaciosystem) [inline src/components/chat/eventUser/index.js]: 👍

File: src/components/chat/message/index.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import React, {Component} from 'react'
+
+export default class Message extends Component{
```
Comment (horaciosystem) [inline src/components/chat/message/index.js]: If this component will have just the render function, you can change it to a `Functional Component`

**Comment by horaciosystem** [comment]
Some tips for your commit messages: [Writing-good-commit-messages](https://github.com/erlang/otp/wiki/Writing-good-commit-messages#do)

File: src/components/chat/eventUser/index.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+      // trata se alguma das promises falhar
+      console.error('Failed retrieving information', err);
+    });
+  }
+  render(){
+    return (
+      <div className="userField">
+        {
+          this.state.chat.map(message => 
+            <Message infos={message} />
```
Comment (horaciosystem) [inline src/components/chat/eventUser/index.js]: When creating items dynamically, you **must** provide an unique `key` property for each item.
> The key is what react uses in the `reconciliation algorithm`, to keep track of all items changes.

File: src/components/chat/eventUser/index.js (hunk 2 above)
Comment (edwinreuleneto) [reply src/components/chat/eventUser/index.js]: Alterado.

File: src/components/chat/message/index.js (hunk 6 above)
Comment (jonathanrz) [reply src/components/chat/message/index.js]: Done in https://github.com/Helabs/rubens-landing-page/pull/3

File: src/components/chat/eventUser/index.js (hunk 3 above)
Comment (jonathanrz) [reply src/components/chat/eventUser/index.js]: Done here: https://github.com/Helabs/rubens-landing-page/commit/ae6dcb05f054154d7a31ebb85413c507c3d3f2aa

**Pull request merged**